files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js

```diff
@@ -3072,18 +3075,17 @@ class CinnamonMenuApplet extends Applet.TextIconApplet {
                 });
             }
         } else if (apps) {
-            for (let i = 0; i < this._applicationsButtons.length; i++) {
-                let button = this._applicationsButtons[i];
+            this._applicationsButtons.forEach(button => {
                 let appId = button.app.get_id();
-                if (apps.indexOf(appId) !== -1) {
+                if (apps.has(appId)) {
```

Comment:
There is no Array.prototype.has, you probably meant to use `includes`.

`Cjs-Message: 19:38:16.555: JS WARNING: [/usr/share/cinnamon/js/ui/lookingGlass.js line 127 > eval 1]: reference to undefined property "has"`

Reply:
Huh? Why would it think this is an Array? I have changed the type of apps from Array to Set so I don't understand this warning.

Reply:
Ah, I see - still doesn't seem necessary - arrays are faster.

Reply:
@LingMan The current logic only allows to show/hide an item, so ordering is not possible at all. In that case, a Set felt semantically the most correct choice. Also, I just assumed that a Set is naturally the fastest in looking up items. That's not really the case in JS...